freeipa/src/main/java/com/sequenceiq/freeipa/service/freeipa/user/VirtualGroupSyncService.java

```diff
@@ -0,0 +1,52 @@
+package com.sequenceiq.freeipa.service.freeipa.user;
+
+import java.util.Map;
+import java.util.Set;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Future;
+import java.util.stream.Collectors;
+
+import javax.inject.Inject;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Qualifier;
+import org.springframework.stereotype.Service;
+
+import com.sequenceiq.cloudbreak.auth.altus.UmsVirtualGroupRight;
+import com.sequenceiq.cloudbreak.auth.altus.VirtualGroupService;
+import com.sequenceiq.freeipa.configuration.UsersyncConfig;
+import com.sequenceiq.freeipa.service.freeipa.user.model.UserSyncOptions;
+
+@Service
+public class VirtualGroupSyncService {
+
+    private static final Logger LOGGER = LoggerFactory.getLogger(VirtualGroupSyncService.class);
+
+    @Inject
+    @Qualifier(UsersyncConfig.USERSYNC_TASK_EXECUTOR)
+    private ExecutorService asyncTaskExecutor;
+
+    @Inject
+    private VirtualGroupService virtualGroupService;
+
+    public void syncVirtualGroups(String accountId, Set<String> environmentCrns, UserSyncOptions options) {
```

Comment:
this method name (and I think the class name) is a bit misleading, as I see only virtual group creation here, and this action doesn't happen in FreeIPA, but in UMS. I think it would be better to rename the class and method eg: `UmsVirtualGroupCreateService` and `createVirtualGroups`

Reply:
ack